lib/homepage.dart

```diff
@@ -62,31 +69,25 @@ class _MyHomePageState extends State<MyHomePage> {
   Future<void> getEvents() async {
     final allEvents = await WebAPI.getEvents();
     if (this.mounted) {
-      setState(() {
```

Comment:
I removed all `setState` calls from individual API calls and called it once all API calls are complete, this is to prevent breaking the app when `setState` is called while the `FutureBuilder` is building.